src/dtls13.c

```diff
@@ -0,0 +1,996 @@
+}
+
+WOLFSSL_METHOD* wolfDTLSv1_3_server_method_ex(void* heap)
+{
+    WOLFSSL_METHOD* method;
+
+    WOLFSSL_ENTER("DTLSv1_3_server_method_ex");
+
+    method = (WOLFSSL_METHOD*) XMALLOC(sizeof(WOLFSSL_METHOD),
+                                  heap, DYNAMIC_TYPE_METHOD);
+    if (method) {
+            InitSSL_Method(method, MakeDTLSv1_3());
+            method->side = WOLFSSL_SERVER_END;
+    }
+
+    return method;
+}
+static int dtls13_rl_add_plaintext_header(WOLFSSL *ssl,
+                                          byte *out,
+                                          enum ContentType content_type,
+                                          size_t length)
+{
+    struct dtls13_record_plaintext_header *hdr;
+
+    hdr = (struct dtls13_record_plaintext_header *)out;
+    hdr->content_type = content_type;
+    hdr->legacy_version_record.major = DTLS_MAJOR;
+    hdr->legacy_version_record.minor = DTLSv1_2_MINOR;
+
+    /* writeSeq updates both epoch and seq */
+    WriteSEQ(ssl, CUR_ORDER, hdr->epoch);
+    c16toa(length, hdr->length);
+
+    return 0;
+}
+
+static int dtls13_handshake_add_header_frag(WOLFSSL *ssl,
+                                            byte *output,
+                                            enum HandShakeType msg_type,
+                                            word32 frag_offset,
+                                            word32 frag_length,
+                                            word32 msg_length)
+{
+    struct dtls13_handshake_header *hdr;
+
+    hdr = (struct dtls13_handshake_header *)output;
+
+    hdr->msg_type = msg_type;
+    c32to24((word32)msg_length, hdr->length);
+    c16toa(ssl->keys.dtls_handshake_number, hdr->message_seq);
+
+    c32to24(frag_offset, hdr->fragment_offset);
+    c32to24(frag_length, hdr->fragment_length);
+
+    return 0;
+}
+
+static int dtls13_type_is_encrypted(enum HandShakeType hs_type)
+{
+    switch (hs_type) {
+    case hello_request:
+    case hello_verify_request:
+    case client_hello:
+    case server_hello:
+        return 0;
+    case encrypted_extensions:
+    case session_ticket:
+    case end_of_early_data   :
+    case hello_retry_request :
+    case certificate         :
+    case server_key_exchange :
+    case certificate_request :
+    case server_hello_done   :
+    case certificate_verify  :
+    case client_key_exchange :
+    case finished            :
+    case certificate_status  :
+    case key_update          :
+    case change_cipher_hs    :
+    case message_hash        :
+    case no_shake            :
+        return 1;
+    }
+
+    /* abide compilers */
+    return 0;
+}
+
+
+static int dtls13_get_rn_mask(WOLFSSL *ssl, byte *ciphertext, byte *mask,
+                              word16 side)
+{
+    byte *key;
+    Aes aes;
+    int ret;
+
+    /* FIXME: implement other ciphers */
+    if (ssl->specs.bulk_cipher_algorithm != wolfssl_aes_gcm)
+        return WOLFSSL_NOT_IMPLEMENTED;
+
+    ret = wc_AesInit(&aes, ssl->heap, -1);
```

Comment:
Use `INVALID_DEVID` here.

Reply:
In the next PR the Aes object will be included in a dedicated Cipher in the SSL object.